packages/esm-patient-banner-app/src/banner-tags/print-identifier-sticker.component.tsx

```diff
@@ -0,0 +1,198 @@
+  const handleBeforeGetContent = useCallback(() => {
+    return new Promise<void>((resolve) => {
+      if (patient && headerTitle) {
+        onBeforeGetContentResolve.current = resolve;
+        setIsPrinting(true);
+        const printStyles = `@media print { @page { size: ${printIdentifierStickerSize}; } }`;
+
+        const style = document.createElement('style');
+        style.appendChild(document.createTextNode(printStyles));
+
+        document.head.appendChild(style);
+      }
+    });
+  }, [patient, printIdentifierStickerSize, headerTitle]);
+
+  const handleAfterPrint = useCallback(() => {
+    onBeforeGetContentResolve.current = null;
+    setIsPrinting(false);
+    closeModal();
+  }, [closeModal]);
+
+  const handlePrint = useReactToPrint({
+    content: () => contentToPrintRef.current,
+    documentTitle: `${patientDetails.name} - ${headerTitle}`,
+    onBeforeGetContent: handleBeforeGetContent,
+    onAfterPrint: handleAfterPrint,
+    onPrintError: (errorLocation, error) => {
+      new Promise<void>((resolve) => {
+        if (error) {
+          if (printRetryCount < 2) {
+            setprintRetryCount(printRetryCount + 1);
+            handleBeforeGetContent().then(() => {
+              handlePrint();
+            });
+          } else {
+            showSnackbar({
+              title: t('printFailed', 'Print Failed'),
+              subtitle: `${t('printingFailed', 'Printing has failed after 3 attempts')}`,
+              kind: 'error',
+              isLowContrast: true,
+            });
+            setprintRetryCount(1);
+          }
+          resolve();
+        }
+      });
+    },
+  });
+
+  const renderElementsInPairs = (elements) => {
+    const pairs = [];
+    let currentPair = [];
+
+    const getKey = (element) => {
+      if (element?.props?.children?.key?.startsWith('identifier-text')) {
+        return 'identifier';
+      }
+      return element?.key;
+    };
+
+    const filteredElements = elements.filter((element) => {
+      return printIdentifierStickerFields.includes(getKey(element));
+    });
+
+    filteredElements.forEach((element, index) => {
+      if (element) {
+        currentPair.push(element);
+        if (currentPair.length === 2) {
+          pairs.push(currentPair);
+          currentPair = [];
+        }
+      }
+    });
+
+    if (currentPair.length === 1) {
+      pairs.push(currentPair);
+    }
+
+    return pairs;
+  };
+
+  return (
+    <div>
+      <ModalHeader closeModal={closeModal} title={t('printIdentifierSticker', 'Print identifier sticker')} />
+      <ModalBody>
+        <div className={styles.stickerContainer} ref={contentToPrintRef}>
+          {printIdentifierStickerFields.includes('name') && (
+            <div key="name" className={styles.patientName}>
+              {patientDetails?.name}
+            </div>
+          )}
+          {renderElementsInPairs(
+            [
+              patientDetails?.identifiers?.map((identifier, index) => (
```

Comment:
It seems like we should pre-check whether there are any identifiers before calling `renderElementsInPairs()` (or its component replacement).